### ISSUE TITLE
Reconnect: g_date_time_compare in sv_ev_room_history 

### ISSUE THREAD

**Issue by DebXWoody** [profanity-im/profanity#1328]
Maybe there is a problem with the implementation of:

`gboolean younger = g_date_time_compare(mucwin->last_msg_timestamp, message->timestamp) < 0 ? TRUE : FALSE;`

in src/event/server_events.c.

After a reconnect there may one of this parameter NULL.

## Steps to Reproduce (for bugs)
Not sure yet.

## Environment
Profanity, version 0.8.1dev.bughunter.d20b83cb

Build information:
XMPP library: libstrophe
Desktop notification support: Enabled
OTR support: Disabled
PGP support: Enabled (libgpgme 1.12.0)
OMEMO support: Enabled
C plugins: Enabled
Python plugins: Enabled (2.7.16)
GTK icons: Enabled

**Comment by jubalh** [comment]
> After a reconnect there may one of this parameter NULL.

Reconned like when using the following?
```
/connect account1
/disconnect
/connect account2
```

Seems not NULL to me.
Can you explain why you think so?

**Comment by DebXWoody** [comment]
I saw some warnings in the logfile.

**Comment by jubalh** [comment]
I tested this with a regular disconnect and with loosing connection. Nothing is every NULL for me.
In the past we set the message->timestamp later/not in call cases, so it could be that you tested with such a version.

But even if this is the case and one of the parameters would be NULL, g_date_time_compare() would return 0 and the behaviour would be as expected.
Just a warming in the log would be printed.
